test-three/amy-menu/src/menu.tsx

```diff
@@ -0,0 +1,93 @@
+import { Component } from "react";
+import menuJson from "./menu-data.json";
+
+type MenuItemProps = {
+  name: string;
+};
+
+class MenuSubItem extends Component<MenuItemProps> {
+  render() {
+    return (
+      <li>
+        {this.props.name}
+      </li>
+    );
+  }
+}
+
+class MenuItem extends Component<MenuItemProps> {
+  render() {
+    return (
+      <ul>
+        {this.props.name}
+        {this.props.children}
+      </ul>
+    );
+  }
+}
+
+export class Menu extends Component {
+  parseJsonFile() {
+    const menuJsonText = JSON.stringify(menuJson);
+    const menuJsonObject = JSON.parse(menuJsonText);
+    return menuJsonObject;
+  }
+
+  createMenuRender() {
+    const menuJsonObject = this.parseJsonFile();
+    const menuRender = menuJsonObject.menu.map((menuItem: any) => {
+      return this.createMenuItems(menuItem);
+    });
+
+    return menuRender;
+  }
+
+  createMenuItems(menuItem: any): JSX.Element {
```

Comment:
Would there be some way to simply this and the `findChildren` function below?

Reply:
It is possible to recursively render React components within themselves. Now that I know there is no ‘header’ list item using between the unordered list element, MenuItem can render another MenuItem if it has any children rather than push into an array. Using a map to go through each menu item, it is possible to render recursively by returning in MenuItem's render something like: {menuItem.children && <MenuItem menuItem ={menuItem.children} />}